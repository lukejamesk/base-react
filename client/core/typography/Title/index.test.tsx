import React from 'react'
import { mount } from 'enzyme'
import { Title } from '.'
import { forEach } from 'ramda'

const titleTypes = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']

describe('<Title />', () => {
  forEach(title => {
    it(`should render with: ${title}`, () => {
      const wrapper = mount(<Title type={title}>My content for {title}</Title>)
      expect(wrapper).toMatchSnapshot()
    })
  }, titleTypes)
})
