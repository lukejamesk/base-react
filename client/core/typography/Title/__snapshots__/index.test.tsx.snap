// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render with: h1 1`] = `
<ForwardRef
  type="h1"
>
  <h1>
    My content for 
    h1
  </h1>
</ForwardRef>
`;

exports[`<Title /> should render with: h2 1`] = `
<ForwardRef
  type="h2"
>
  <h2>
    My content for 
    h2
  </h2>
</ForwardRef>
`;

exports[`<Title /> should render with: h3 1`] = `
<ForwardRef
  type="h3"
>
  <h3>
    My content for 
    h3
  </h3>
</ForwardRef>
`;

exports[`<Title /> should render with: h4 1`] = `
<ForwardRef
  type="h4"
>
  <h4>
    My content for 
    h4
  </h4>
</ForwardRef>
`;

exports[`<Title /> should render with: h5 1`] = `
<ForwardRef
  type="h5"
>
  <h5>
    My content for 
    h5
  </h5>
</ForwardRef>
`;

exports[`<Title /> should render with: h6 1`] = `
<ForwardRef
  type="h6"
>
  <h6>
    My content for 
    h6
  </h6>
</ForwardRef>
`;
